nifi-nar-bundles/nifi-influxdb-bundle/nifi-influxdb-processors/src/main/java/org/apache/nifi/processors/influxdb/ExecuteInfluxDBQuery.java

```diff
@@ -0,0 +1,258 @@
+
+            outgoingFlowFile = incomingFlowFile;
+        } else {
+            charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions().getValue());
+            query = context.getProperty(INFLUX_DB_QUERY).evaluateAttributeExpressions().getValue();
+            database = context.getProperty(DB_NAME).evaluateAttributeExpressions().getValue();
+            queryResultTimeunit = TimeUnit.valueOf(context.getProperty(INFLUX_DB_QUERY_RESULT_TIMEUNIT).evaluateAttributeExpressions().getValue());
+
+            outgoingFlowFile = session.create();
+        }
+
+        try {
+            long startTimeMillis = System.currentTimeMillis();
+            QueryResult result = executeQuery(context, database, query, queryResultTimeunit);
+
+            if ( result == null ) {
+                String message = "Query Result was " + result;
+                getLogger().error(message);
+                throw new NullPointerException(message);
+            }
+
+            String json = gson.toJson(result);
+
+            if ( getLogger().isDebugEnabled() ) {
+                getLogger().debug("Query result {} ", new Object[] {result});
+            }
+
+            ByteArrayInputStream bais = new ByteArrayInputStream(json.getBytes(charset));
+            session.importFrom(bais, outgoingFlowFile);
+            bais.close();
+
+            final long endTimeMillis = System.currentTimeMillis();
+
+            if ( ! result.hasError() ) {
+                outgoingFlowFile = session.putAttribute(outgoingFlowFile, INFLUX_DB_EXECUTED_QUERY, String.valueOf(query));
+                session.transfer(outgoingFlowFile, REL_SUCCESS);
+            } else {
+                outgoingFlowFile = populateErrorAttributes(session, outgoingFlowFile, query, result.getError());
+                session.transfer(outgoingFlowFile, REL_FAILURE);
+            }
+
+            session.getProvenanceReporter().send(outgoingFlowFile, makeProvenanceUrl(context, database),
+                (endTimeMillis - startTimeMillis));
+        } catch (Exception exception) {
+            outgoingFlowFile = populateErrorAttributes(session, outgoingFlowFile, query, exception.getMessage());
+            if ( exception.getCause() instanceof SocketTimeoutException ) {
+                getLogger().error("Failed to read from influxDB due SocketTimeoutException to {} and retrying",
+                        new Object[]{exception.getCause().getLocalizedMessage()}, exception.getCause());
+                session.transfer(outgoingFlowFile, REL_RETRY);
+            } else {
+                getLogger().error("Failed to read from influxDB due to {}",
+                        new Object[]{exception.getLocalizedMessage()}, exception);
+                session.transfer(outgoingFlowFile, REL_FAILURE);
+            }
+            context.yield();
+        }
+    }
+
+    protected String makeProvenanceUrl(final ProcessContext context, String database) {
```

Comment:
Adding a flowfile parameter here would help with the EL support.

Reply:
Added flow file parameter.